### DISCUSSION TITLE
Cannot link Transfer to Bill - Intended Behavior?

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [firefly-iii/firefly-iii#8792] [Firefly III] [answered]
I have a bill and associated rule set up to handle monthly payments to my credit card. When I make a payment however, the payment is not associated with the bill and the bill shows as unpaid in my billing screen.

When running a test on the rule, the query returns three results, but when manually running the rule (or going into the bill and searching for matching loads), the transfer is never found. Editing the transfer itself shows that the associated bill field is missing.

This seems to be a consequence of how the account is set up - I followed the developer's recommendation to make a standard account for my credit card, and then to make transfers to this account to "top up" my balance rather than treating it as a bill.

Regardless of setup however, this is a regular monthly expense for me and for all practical purposes I treat it as a bill as far as making regular payments, and as such I feel it would be helpful to be able to treat it as a bill.

**Comment by JC5** [comment, marked as answer]
This is correct, bills are not linked to transfers, but to withdrawals only. I use a piggy bank personally, to save for the credit card payment (which is also a transfer in my setup)

**Reply by TheBrambleShark** [reply]
I'll look at using the piggy bank here, thanks!

**Comment by PeteS4** [comment]
Hijacking this question as I was wondering the same thing, in a slightly different context:
I have a monthly fund savings plan, the savings of which I also track in FFIII as a savings account. Some fixed amount is automatically transferred every month and I wanted to use a bill for that, just so that I don't forget and accidentally overspend. I see how a piggy bank can be used for credit card payments which are just a "delayed withdrawal" of payments that already occured, but I think my situation is a bit different conceptionally. Any advice on how to handle that?

**Reply by zebrapurring** [reply]
I have exactly the same situation, I still created a subscription (formerly called bills) for my monthly contributions just so I can see the expected monthly costs. However it's annoying that we cannot link transactions to a subscription, so they always show up as unpaid...
